File: frontend/src/types/vote.ts
export type ContentType = 'news' | 'price' | 'ai' | 'meme';

export interface VoteCreate {
  content_type: ContentType;
  content_id: string;
  is_upvote: boolean;
}

export interface Vote {
  id: string;
  user_id: string;
  content_type: ContentType;
  content_id: string;
  is_upvote: boolean;
  created_at: string;
}

export interface VoteResponse {
  success: boolean;
  data: Vote;
}

export interface VotesResponse {
  success: boolean;
  data: Vote[];
}

File: backend/app/schemas/vote.py
"""Schemas for voting functionality."""
from pydantic import BaseModel, Field
from typing import Literal
from datetime import datetime
from uuid import UUID


class VoteCreate(BaseModel):
    """Schema for creating a vote."""
    content_type: Literal["news", "price", "ai", "meme"] = Field(
        ...,
        description="Type of content being voted on"
    )
    content_id: str = Field(
        ...,
        description="Identifier for the specific content item"
    )
    is_upvote: bool = Field(
        ...,
        description="True for upvote, False for downvote"
    )


class VoteResponse(BaseModel):
    """Schema for vote response."""
    id: UUID
    user_id: UUID
    content_type: str
    content_id: str
    is_upvote: bool
    created_at: datetime

    class Config:
        from_attributes = True


class VoteUpdate(BaseModel):
    """Schema for updating a vote."""
    is_upvote: bool = Field(
        ...,
        description="True for upvote, False for downvote"
    )

